rotkehlchen/premium/sync.py

```diff
@@ -115,22 +115,26 @@ def _sync_data_from_server_and_replace_local(self) -> Tuple[bool, str]:
 
         return True, ''
 
-    def maybe_upload_data_to_server(self, force_upload: bool = False) -> bool:
+    def check_if_should_sync(self, force_upload: bool) -> bool:
         # if user has no premium do nothing
         if self.premium is None:
             return False
 
-        with self.data.db.user_write() as cursor:
+        with self.data.db.conn.read_ctx() as cursor:
             if not self.data.db.get_setting(cursor, 'premium_should_sync') and not force_upload:
                 return False
 
-            # upload only once per hour
-            diff = ts_now() - self.last_data_upload_ts
-            if diff < 3600 and not force_upload:
-                return False
+        # upload only once per hour
+        diff = ts_now() - self.last_data_upload_ts
+        if diff < 3600 and not force_upload:
+            return False
 
+        return True
+
+    def maybe_upload_data_to_server(self, force_upload: bool = False) -> bool:
+        with self.data.db.user_write() as cursor:
             try:
-                metadata = self.premium.query_last_data_metadata()
+                metadata = self.premium.query_last_data_metadata()  # type: ignore
```

Comment:
why this type ignore? Add an `assert premium is not None, 'call should make sure premium exists` at start instead.

This will be optimized out for production and make it obvious to the reader.